### PR TITLE
[Feature] Added resource `databricks_custom_app_integration`

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes
Add resource `databricks_custom_app_integration` for OAuth custom app integration

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: docs/resources/custom_app_integration.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+* `integration_id` - Unique integration id for the custom OAuth app.
+* `client_id` - OAuth client-id generated by Databricks
+* `client_secret` - OAuth client-secret generated by the Databricks if this is a confidential OAuth app.
+
+## Import
+
+This resource can be imported by Databricks account ID and integration ID.
+
+```sh
+terraform import databricks_budget.this '<account_id>|<budget_configuration_id>'
```
Comment (alexott) [inline docs/resources/custom_app_integration.md:50]: we need to fix copy/paste :-)

File: apps/resource_custom_app_integration.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	ClientSecret string `json:"client_secret,omitempty"`
+}
+
+func ResourceCustomAppIntegration() common.Resource {
+	s := common.StructToSchema(CustomAppIntegration{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
+		for _, p := range []string{"client_id", "create_time", "created_by", "creator_username", "integration_id"} {
+			common.CustomizeSchemaPath(m, p).SetComputed()
+		}
+		for _, p := range []string{"name", "scopes"} {
+			common.CustomizeSchemaPath(m, p).SetForceNew()
```
Comment (alexott) [inline apps/resource_custom_app_integration.go:24]: Can we update `confidential`? If yes, then we may need to use force send fields in the update request if we can change from true to false. If no, then we need to mark it as force new

**Review by alexott — APPROVED** [review]

**Comment by alexott** [comment]
`CustomAppIntegrationUpdate` test is failing

**Comment by nkvuong** [comment]
integration tests passed (failed tests are unrelated as they are on workspace-level tests)

**Pull request merged**